### PR TITLE
Some minor bug fixes & Add basic auth support 

### PR DESCRIPTION
**Pull request by capraynor**
Fix bugs: 
1. pyperclip throws exception on linux CLI without x window. 

Add Feature: 
Add http basic auth support.

### CODE REVIEW

File: scraper/src/documentation_spider.py [hunk 1]
```diff
@@ -24,6 +24,8 @@ class DocumentationSpider(CrawlSpider, SitemapSpider):
     """
     DocumentationSpider
     """
+    http_user=os.environ.get('DOCSEARCH_BASICAUTH_USERNAME', None)
```
Comment (s-pace) [inline scraper/src/documentation_spider.py]: What is the use of these vars?

Comment (capraynor) [reply scraper/src/documentation_spider.py]: These vars will provide HTTP basic auth username and password. 
I think that these vars should be provided in config file, but these are passwords, so better read the information from environment variables.

Comment (capraynor) [reply scraper/src/documentation_spider.py]: Sorry, I think that I'm misunderstanding your question. The "http_user" is provided by scrapy.

**Comment by capraynor** [comment]
> Does it solve #440 too?
> LGTM

Yes of course @s-pace 

This fixes #440. 

Here's my thought: 
At first, I tried to fix this by adding a HTTP header, but with no luck.
And then, I found that scrapy provides http_user to perform HTTP basic auth. It is a plugin. 
So I just read a environment variable, and put that environment variable to HTTP basic auth header. 

I have tried that if both env var is None, It will not throw exception. (Null Check). 

IMHO, #440 is a little heavy :-D

**Comment by s-pace** [comment]
THX @capraynor

**Pull request merged**